feat: add a new file to ltips for candidates (#12)
loft-br/hiring-process-study-guide: software-engineer/pt_BR/do-and-donts.md

# Dicas do que fazer e o que não fazer durante as entrevistas

Esta é uma lista direta e destilada de o que fazer e não fazer em entrevistas técnicas, principalmente para entrevistas algorítmicas. Alguns deles podem se aplicar apenas a entrevistas or vídeo ou entrevistas no quadro branco, mas a maioria se aplica a ambos. 

#### Lenda

- ✅ = Faça
- ❌ = Não
- ⚠️ = Situacional

### Antes da entrevista

| | Coisas |
| --- | --- |
| ✅ | Prepare caneta, papel e fones de ouvido / fones de ouvido. |

> | ✅ | Encontre um ambiente silencioso com boa conexão à Internet. |

Reviewer: Mixed feelings sobre essa frase. Muitas pessoas não possuem um ambiente silencioso em casa ou uma conexão boa a internet. Tipo, quem tem filho(s) pequenos em casa ou uma obra próxima ou mesmo outras pessoas trabalhando de casa, no mesmo espaço, pode ter dificuldades com encontrar um ambiente silencioso... Da mesma forma, a conexão a internet é a que a pessoa já possui, a gente dá um jeito de seguir se ela não for perfeita (tipo desligando câmeras...)

| ✅ | Verifique se a webcam e o áudio estão funcionando. As entrevistas, em geral, serão realizadas pelo Zoom. Confira se eles esta instalado e configurado corretamente.|

> | ✅ | Decida sobre uma linguagem de programação e familiarize-se com ela. |

Reviewer: ```suggestion
| ✅ | Escolha a linguagem de programação que você tiver maior familiaridade. Não temos absolutamente nenhuma restrição nesse ponto!|
```
Eu só deixaria mais claro aqui que não temos restrições quanto à linguagem escolhida.

| ✅ | Familiarize-se com o ambiente de codificação (CoderPad / CodePen / VSCode). Configure os atalhos de codificação, ative o preenchimento automático, espaçamento de tabulação, etc.

> | ✅ | Prepare algumas perguntas para sem feitas no final da entrevista. |

Reviewer: concordo com o comentário acima... Talvez o texto abaixo possa deixar isso melhor:
```suggestion
| ✅ | Deixamos sempre 5 ou 10 minutos ao final de cada entrevista para responder perguntas de quem está sendo entrevistado. Caso tenha alguma coisa que queira saber sobre a empresa, rotina ou detalhes técnicos, é um excelente momento para perguntar isso!|
```

| ✅ | Vista-se confortavelmente. Normalmente você não precisa usar roupas elegantes, o casual deve servir. |

> | ✅ | Fique calmo e composto. |

Author: Acho que podemos deixar uma mensagem tranquilizante, neste caso.

> | ⚠️ | Desligue a webcam, se possível. A maioria das entrevistas remotas não exigirá bate-papo por vídeo e deixá-lo ligado serve apenas como uma distração. |

Reviewer: Acho que a câmera ligada transmite mais pessoalidade para a entrevista. Eu particularmente fico um pouco incomodado quando a pessoa que está sendo entrevistada não liga a cam e não fala nada sobre isso.. Talvez mudando para o texto abaixo:
```suggestion
| ⚠️ | Dê preferência por deixar a webcam ligada. Caso isso não seja possível ou seja desconfortável para você, zero problemas, só fale com a pessoa entrevistadora sobre isso.|
```

### Introdução

| | Coisas |
| --- | --- |
| ✅ | Apresente-se em algumas frases em um ou dois minutos. |
| ✅ | Mencione pontos interessantes que sejam relevantes para a função para a qual você está se candidatando. |

> | ✅ | Pareça entusiasmado! Fale com um sorriso e você naturalmente parecerá mais envolvente. |

Reviewer: Outro ponto que pensei aqui sobre essa frase, muitas vezes a pessoa que está sendo entrevistada não buscou ativamente a Loft, mas foi recrutada por alguma pessoa aqui de dentro (EM ou Tech Recruiter). Acho que talvez esse tipo de pessoa não esteja de fato entusiasmada para trabalhar aqui, nós que estamos entusiasmados com a possibilidade de contar com ela aqui!

> | ✅ | Pareça entusiasmado! Fale com um sorriso e você naturalmente parecerá mais envolvente. |

Author: E discordo do ponto de se o candidato deve ou não estar entusiasmado. E tendo a pensar que se ela esta passando por um processo seletivo obviamente deve estar entusiasmado com a possibilidade de ser contratado para a vaga. Agora quanto a questão do sorriso, acho que não é necessário, realmente.

> | ✅ | Pareça entusiasmado! Fale com um sorriso e você naturalmente parecerá mais envolvente. |

Author: Acho que uma frase melhor seria: Demonstre entusiasmo e você naturalmente parecerá mais envolvente.

| ❌ | Passe muito tempo se apresentando. Quanto mais tempo você gasta falando, menos tempo você tem para codificar. |

### Ao receber a pergunta

| | Coisas |
| --- | --- |

> | ✅ | Repita a pergunta para o entrevistador. |

Reviewer: ```suggestion
| ✅ | Repita a pergunta para quem estiver te entrevistando, para confirmar que o entendimento está correto. |
```
Neutralidade de Gênero

> | ✅ | Esclareça quaisquer suposições que pareça inconscientemente. Muitas questões são subespecificadas propositalmente. Por exemplo, um diagrama em forma de árvore poderia muito bem ser um gráfico que permite ciclos e uma solução recursiva ingênua não funcionaria. |

Reviewer: Tenho a impressão de que esse ponto é mais voltado para entrevistas de algoritmos, tipo qual algoritmo ou estrutura de dados utilizar. Talvez para a nossa entrevista ele não faça tanto sentido.

| ✅ | Esclareça o formato e o intervalo de entrada. Pergunte se a entrada pode ser considerada bem formada e não nula. |
| ✅ | Trabalhe com um pequeno exemplo para garantir que você entendeu a pergunta. |
| ✅ | Explique uma abordagem de alto nível, mesmo que seja de força bruta. |
| ✅ | Melhore a abordagem e otimize. Reduza o trabalho duplicado e armazene em cache cálculos repetidos. |

> | ✅ | Pense com cuidado, a seguir declare e explique a complexidade de tempo e espaço de suas abordagens. |

Reviewer: Eu removeria essa parte de tempo de complexidade de tempo e espaço, foi um dos pontos que a gente quis resolver com a prova nova de coding

> | ✅ | Se estiver travado, pense nos problemas relacionados que você viu antes e como eles foram resolvidos. |

Reviewer: ```suggestion
| ✅ | Se estiver com dificuldades em prosseguir, pense nos problemas relacionados que você viu antes e como eles foram resolvidos. |
```
Neutralidade de gênero

| ❌ | Ignore as informações fornecidas a você. Cada peça é importante. |
| ❌ | Passe para a codificação imediatamente. |

> | ❌ | Comece a codificar sem a luz verde do entrevistador. |
> | ❌ | Pareça muito inseguro sobre sua abordagem ou análise. |

Reviewer: ```suggestion
| ❌ | Comece a codificar sem a luz verde da pessoa entrevistadora. |
| ❌ | Demonstre muita insegurança sobre sua abordagem ou análise. |
```
Neutralidade de gênero

### Codifique sua solução

| | Coisas |
| --- | --- |

> | ✅ | Explique o que você está codificando / digitando para o entrevistador, o que você está tentando alcançar. |

Reviewer: ```suggestion
| ✅ | Explique o que você está codificando / digitando para a pessoa entrevistadora. Deixe claro o que você está tentando alcançar. |
```
Neutralidade de gênero

| ✅ | Pratique um bom estilo de codificação. Nomes de variáveis ​​claros, espaçamento de operador consistente, recuo adequado, etc.
| ✅ | Digite / escreva a uma velocidade razoável. Muito lento não é bom. |
| ✅ | Tanto quanto possível, escreva código compilável real, não pseudocódigo. |
| ✅ | Escreva de forma modular. Extraia pedaços de código repetido em funções. |
| ✅ | Peça permissão para usar funções triviais sem ter que implementá-las; você economiza algum tempo. |

> | ✅ | Use as dicas do entrevistador. |

Reviewer: ```suggestion
| ✅ | Use as dicas da pessoa entrevistadora. |
```
Neutralidade de gênero, concordo totalmente com subir a prioridade dela

| ✅ | Demonstre domínio da linguagem de programação escolhida. |
| ✅ | Demonstrar conhecimento técnico em estruturas de dados e algoritmos. |

> | ✅ | Se você estiver abrindo caminho em seu código, diga isso em voz alta para o seu entrevistador e diga o que você faria em um ambiente sem entrevista (sem restrições de tempo). Por exemplo, "Em configurações de não entrevista, eu escreveria um regex para analisar essa string em vez de usar` split () `que pode não cobrir todos os casos." |

Reviewer: ```suggestion
| ✅ | Se você estiver abrindo caminho em seu código, diga isso em voz alta para quem estiver te entrevistando e diga o que você faria em um ambiente sem entrevista (sem restrições de tempo). Por exemplo, "Em configurações de não entrevista, eu escreveria um regex para analisar essa string em vez de usar` split () `que pode não cobrir todos os casos." |
```
Neutralidade de gênero

| ✅ | Pratique habilidades de gerenciamento de espaço em quadro branco. |
| ⚠️ | Codificação defensiva razoável. Verifique se há nulos, coleções vazias, etc. Pode omitir se a validade da entrada foi esclarecida com o entrevistador. |

> | ❌ | Fique quieto o tempo todo enquanto codifica. |

Reviewer: ```suggestion
| ❌ | Fique em silêncio enquanto codifica. |
```
Neutralidade de gênero

| ❌ | Gaste muito tempo escrevendo comentários. |
| ❌ | Use nomes de variáveis ​​extremamente verbosos ou de caractere único (a menos que sejam comuns como `i`,` n`). |
| ❌ | Copie e cole o código sem verificar (por exemplo, as variáveis ​​precisam ser renomeadas). |

> | ❌ | Interrompa o seu entrevistador quando ele estiver falando. Normalmente, se eles falam, estão tentando dar dicas ou guiá-lo na direção certa. |

Reviewer: ```suggestion
| ❌ | Interrompa a pessoa entrevistadora quando ela estiver falando. Normalmente, se elas falam, estão tentando dar dicas ou guiar você na direção certa. |
```
Neutralidade de gênero

| ❌ | Escreva muito grande (ocupa muito espaço) ou muito pequeno (ilegível) se estiver em um quadro branco. |

### Após a codificação

| | Coisas |
| --- | --- |
| ✅ | Verifique se há erros em seu código, como se fosse a primeira vez que você via um código escrito por outra pessoa. |

> | ✅ | Verifique se há erros individuais. |

Author: acho o tip acima já é o suficiente

| ✅ | Crie casos de teste. Experimente casos de teste extremos - conjuntos vazios, conjuntos de item único, números negativos |
| ✅ | Percorra seu código com esses casos de teste. |
| ✅ | Procure lugares onde você pode refatorar. |

> | ✅ | Reitere a complexidade de tempo e espaço de seu código. |

Reviewer: Também removeria esse ponto, pelo mesmo motivo citado acima, de ser o que buscamos com a mudança da etapa de coding.
```suggestion
```

Author: Eu acho que seria importante esse ponto, mesmo que não tenhamos uma parte mais técnica sobre algoritmos. Quando o candidato demonstra conhecimento sobre complexidade de tempo e espaço, em minha opinião, é um ponto muito positivo.

Reviewer: Concordo contigo, cara, é uma coisa positiva quando a pessoa que está sendo entrevistada fala sobre isso, porém não é algo que nós cobramos ativamente e nem deveria ser um ponto negativo quando a pessoa não fala. Nós tínhamos dois grandes problemas na prova de coding anterior, que era mais voltada para algoritmos, pessoas que gostavam do estilo da prova e se frustravam, dado que aqui na Loft não temos muitos desafios desse tipo, e pessoas que não gostavam e dropavam do processo. Meu receio é com essa frase voltar a incorrer nesses dois pontos.

Author: Acho que poderíamos trocar a frase para algo do tipo: Caso se sinta a vontade e confiante, mencione sobre  a complexidade de tempo e espaço de seu código.

| ✅ | Explique as compensações e como o código / abordagem pode ser melhorado se tiver mais tempo. |
| ❌ | Anuncie imediatamente que você concluiu a codificação. Faça o anterior primeiro! |

> | ❌ | Discuta com o entrevistador. Eles podem estar errados, mas isso é muito improvável, visto que estão familiarizados com a questão. |

Reviewer 2: Acho que dá pra pensar num phrasing menos "intimidador" aqui.
```suggestion
| ❌ | Discuta com o entrevistador. Eles podem estar errados, mas isso é muito improvável, visto que estão familiarizados com a questão. |
```

Reviewer: Eu removeria esse ponto. Acho que discussões entre as pessoas que estão entrevistando e sendo entrevistadas podem ser bem válidas e produtivas em algumas .

Author: Acho que podemos remover esse ponto sim.

> ### Embrulhar

Reviewer 2: ```suggestion
### Finalizando
```

| | Coisas |
| --- | --- |
| ✅ | Pergunte. Mais importante ainda, faça perguntas boas e envolventes, feitas sob medida para a empresa! |

> | ✅ | Agradeça ao entrevistador. |

Reviewer: ```suggestion
```
Removeria esse ponto, nós não estamos fazendo nenhum favor à pessoa que está sendo entrevistada, geralmente eu finalizo a entrevista agradecendo à pessoa entrevistada pelo tempo e disponibilidade de participar do nosso processo seletivo.

> | ⚠️ | Pergunte sobre o seu desempenho na entrevista. Isso pode ficar estranho. |

Reviewer 2: Esse phrasing ficou estranho, e acho que deveria ser um Don't mesmo (não situacional)

Reviewer: ```suggestion
| ❌  | Pergunte sobre o seu desempenho na entrevista. Sempre temos mais de uma pessoa entrevistando, ao término da entrevista essas pessoas se reúnem e trocam impressões para gerar um feedback. Esse feedback é consolidado com os das demais etapas e enviado a você, prometemos que fazemos isso rapidamente! |
```

> | ❌ | Termine a entrevista sem fazer perguntas. |

Reviewer: ```suggestion
```
Removeria esse ponto. Se a pessoa não tem nenhuma dúvida mais para tirar conosco, não vejo nenhum problema em finalizar a entrevista sem perguntas dela.

> ### Postar entrevista

Reviewer 2: ```suggestion
### Após a entrevista
```

| | Coisas |
| --- | --- |
| ✅ | Anote as perguntas e respostas da entrevista, pois podem ser úteis para referência futura. |
| ⚠️ | Envie um e-mail de acompanhamento ao (s) seu (s) entrevistador (es) agradecendo pelo tempo dispensado e pela oportunidade de entrevistá-los. |

> Esse conteúdo utilizou como referência o artigo publicado em [http://techinterviewhandbook.org/cheatsheet/](http://techinterviewhandbook.org/cheatsheet/). Acesse para mais dicas e informações.

Reviewer 2: ```suggestion
Esse conteúdo é uma adaptação [deste artigo](http://techinterviewhandbook.org/cheatsheet/), que pode ser acessado para dicas complementares (principalmente nos demais artigos do mesmo site).
```

> # Dicas do que fazer e o que não fazer durante as entrevistas

Reviewer 3: Pelo que entendi esse doc é focado em dicas para a etapa de Coding Skills. 
É isso mesmo, ou seria para todas as etapas? 🤔

Author: Seria para todas as etapas técnicas.

> Esta é uma lista direta e destilada de o que fazer e não fazer em entrevistas técnicas, principalmente para entrevistas algorítmicas. Alguns deles podem se aplicar apenas a entrevistas or vídeo ou entrevistas no quadro branco, mas a maioria se aplica a ambos. 

Reviewer 4: lista destilada é um tradução de distilled list, destilada em português tem essa significado?

Reviewer 2: boa, alterei minha sugestão acima pra tratar esse ponto

#### Leganda

- ✅ = Faça

> - ❌ = Não

Reviewer 4: Não faça

- ⚠️ = Situacional

### Antes da entrevista

| | Coisas |
| --- | --- |
| ✅ | Prepare caneta, papel e fones de ouvido. |

> | ✅ | Se possível, encontre um ambiente silencioso com boa conexão à Internet. Nós entendemos que nem todos os candidatos tem um ambiente calmo e tranquilo ou mesmo uma boa conexão de internet. |

Reviewer 2: ```suggestion
| ✅ | Dê preferência para um ambiente silencioso e com boa conexão à Internet. Mas se não for possível, tudo bem :) |
```

| ✅ | Verifique se a webcam e o áudio estão funcionando. As entrevistas, em geral, serão realizadas pelo Zoom. Confira se ele está instalado e configurado corretamente.|
| ✅ | Escolha a linguagem de programação que você tiver maior familiaridade. Não temos absolutamente nenhuma restrição nesse ponto! |
| ✅ | Familiarize-se com o ambiente de codificação (CoderPad / CodePen / VSCode). Configure os atalhos de codificação, ative o preenchimento automático, espaçamento de tabulação, etc.

> | ✅ | Deixamos sempre 5 ou 10 minutos ao final de cada entrevista para responder perguntas de quem está sendo entrevistado. Caso tenha alguma coisa que queira saber sobre a empresa, rotina ou detalhes técnicos, é um excelente momento para perguntar isso!|

Reviewer 2: ```suggestion
| ✅ | Deixamos sempre de 5 a 10 minutos no final de cada entrevista para responder perguntas de quem está sendo entrevistado (sobre a empresa, rotina, detalhes técnicos etc). |
```

> | ✅ | Vista-se confortavelmente. Normalmente você não precisa usar roupas elegantes. |

Reviewer 2: ```suggestion
| ✅ | Vista-se confortavelmente. Não é necessário usar roupas elegantes. |
```

> | ✅ | Procure se manter calmo, na medida do possível. Entendemos que uma entrevista não é o ambiente mas confortável do mundo, por isso, vamos te apoiar e tentar trazer o ambiente mais "leve" que pudermos. |

Reviewer 2: ```suggestion
| ✅ | Procure manter a calma, na medida do possível. Entendemos que uma entrevista não é a situação mais confortável do mundo. Por isso, vamos te apoiar e tentar trazer o ambiente mais "leve" que pudermos 🧡 |
```

| ⚠️ | Dê preferência por deixar a webcam ligada. Caso isso não seja possível ou seja desconfortável para você, não tem problema, é só falar com a pessoa entrevistadora sobre isso. |

### Introdução

| | Coisas |
| --- | --- |
| ✅ | Apresente-se brevemente. |

> | ✅ | Mencione pontos interessantes que sejam relevantes para a função para a qual você está se candidatando. |

Reviewer 3: Será que faz sentido a gente dar essa dica? Já peguei caso de pessoas que começam a falar da experiência, e perdem o foco da entrevista.

Em tese essa parte já será coberta na entrevista de background. (a menos que essas dicas também sejam para a etapa de background)

Reviewer: Já tive entrevistas onde a pessoa entrevistada deu voltas e mais voltas na parte de se apresentar. Hoje, não peço mais para que quem eu estiver entrevistando se apresente, temos 3 etapas técnicas no Meeting Loft, acho que fica até maçante para a própria pessoa entrevistada se apresentar em todas as vezes. Acho que talvez possamos remover esses pontos daqui...

Reviewer 2: concordo. acho que podemos reescrever esse bloco pra focar em deixar a apresentação breve e reforçar que o histórico da pessoa será explorado numa etapa posterior

| ✅ | Demonstre entusiasmo e você naturalmente parecerá mais envolvente. |
| ❌ | Passe muito tempo se apresentando. Quanto mais tempo você gasta falando, menos tempo você tem para codificar. |

### Ao receber a pergunta

| | Coisas |
| --- | --- |
| ✅ | Repita a pergunta para quem estiver te entrevistando, para confirmar que o entendimento está correto. |

> | ✅ | Esclareça quaisquer suposições que pareça inconscientemente. Muitas questões são subespecificadas propositalmente. Por exemplo, se perguntado como você implementaria um sistema de upload, questione possíveis "pontas soltas" em uma povável solução para um sistema de upload, como, por exemplo, quais formatos de arquivos serão permitidos. |

Reviewer 2: ```suggestion
| ✅ | Manifeste todas as pressuposições que você for assumir durante o exercício. Algumas questões deixam margem para interpretação de propósito. Por exemplo, se perguntado como você implementaria um sistema de upload, questione possíveis "pontas soltas" em uma provável solução, como, por exemplo, quais formatos de arquivos serão permitidos. |
```

| ✅ | Esclareça o formato e o intervalo de entrada. Pergunte se a entrada pode ser considerada bem formada e não nula. |
| ✅ | Trabalhe com um pequeno exemplo para garantir que você entendeu a pergunta. |
| ✅ | Explique uma abordagem de alto nível, mesmo que seja de força bruta. |
| ✅ | Melhore a abordagem e otimize. Reduza o trabalho duplicado e armazene em cache cálculos repetidos. |
| ✅ | Se estiver com dificuldades em prosseguir, pense nos problemas relacionados que você viu antes e como eles foram resolvidos. |
| ❌ | Ignore as informações fornecidas a você. Cada peça é importante. |
| ❌ | Passe para a codificação imediatamente. |

> | ❌ | Comece a codificar sem a luz verde da pessoa entrevistadora. |

Reviewer 2: ```suggestion
| ❌ | Comece a codificar sem o aval da pessoa entrevistadora. |
```

| ❌ | Demonstre muita insegurança sobre sua abordagem ou análise. |

### Codifique sua solução

| | Coisas |
| --- | --- |
| ✅ | Explique o que você está codificando / digitando para a pessoa entrevistadora. Deixe claro o que você está tentando alcançar. |
| ✅ | Pratique um bom estilo de codificação. Nomes de variáveis ​​claros, espaçamento de operador consistente, recuo adequado, etc.

> | ✅ | Digite / escreva a uma velocidade razoável. Muito lento não é bom. |

Reviewer 3: Acho que esse ponto pode deixar as pessoas mais apreensivas do que o necessário.
Você já teve casos de pessoas que digitavam de forma tão lenta em que isso foi um problema? 🤔

Author: A já peguei casos em que a pessoa foi mal pela demora em escrever ou fazer os diagramas. Acho que é um ponto relevante.

Reviewer: Será que a pessoa foi mal pq demorou para escrever ou desenhar os diagramas ou será que demorou porque não estava tão segura do que estava fazendo, talvez por falta de conhecimentos? Concordo com o Gabriel, acho que esse ponto tende a gerar mais ansiedade do que ajudar as pessoas candidatas.

| ✅ | Tanto quanto possível, escreva código compilável real, não pseudocódigo. |
| ✅ | Escreva de forma modular. Extraia pedaços de código repetido em funções. |

> | ✅ | Peça permissão para usar funções triviais sem ter que implementá-las; você economiza algum tempo. |

Reviewer 3: Não entendi muito bem esse ponto de pedir permissão. 🤔

Reviewer 2: É, acho que cabe um rephrasing. Sugestão:
```suggestion
| ✅ | Você pode usar livremente os recursos nativos da linguagem (e até mesmobibliotecas) para economizar tempo nos exercícios. |
```

| ✅ | Use as dicas da pessoa entrevistadora. |

> | ✅ | Demonstre domínio da linguagem de programação escolhida. |

Reviewer 3: Pra mim essa dica ficou um pouco etérea e pouco prática. Também vejo um risco aqui, de isso fazer as pessoas começarem a trazer soluções maiores do que o necessário, só para mostrar domínio da linguagem.
Faz sentido? 🤔

Reviewer: Concordo com esse ponto. Acho que isso inclusive vai na contra-mão do que falamos no início da entrevista de coding, que é uma dinâmica de pair programming normal e que a pessoa deve ficar totalmente à vontade para fazer buscas no google, por exemplo... 
Acho que já cobrimos o que poderia ser dito por esse ponto quando dissemos lá em cima para a pessoa escolher a linguagem de programação que tenha mais familiaridade, voto por remover esse ponto...

Author: Removed!

> | ✅ | Demonstrar conhecimento técnico em estruturas de dados e algoritmos. |

Reviewer 2: ```suggestion
| ✅ | Demonstre conhecimento técnico em estruturas de dados e algoritmos. |
```

> | ✅ | Se você estiver abrindo caminho em seu código, diga isso em voz alta para quem estiver te entrevistando e diga o que você faria em um ambiente sem entrevista (sem restrições de tempo). Por exemplo, "Em configurações de não entrevista, eu escreveria um regex para analisar essa string em vez de usar` split () `que pode não cobrir todos os casos." |

Reviewer 2: ```suggestion
| ✅ | Se você estiver abrindo caminho em seu código, diga isso em voz alta para quem estiver te entrevistando e diga o que você faria em um ambiente sem entrevista (sem restrições de tempo). Por exemplo, "Em um cenário de não-entrevista, eu escreveria um regex para analisar essa string em vez de usar `split()`, que pode não cobrir todos os casos." |
```

> | ✅ | Pratique habilidades de gerenciamento de espaço em quadro branco. |

Reviewer 3: Será que isso faz diferença real na entrevista? Em tese a gente não deveria estar avaliando o quão bem organizado e "bonito" está o desenho, e sim o que a pessoa trás na fala ao desenhar. 🤔

Reviewer: Acho que isso seria uma questão mais para entrevistas presenciais, com o quadro branco físico. Com o diagrams.net, o espaço do diagrama é infinito, também não vejo isso como uma questão relevante...

Author: Acho válido substituir por algo do tipo: Pratique habilidades desenho de diagramas em ferramentas como o diagrams.net.

Reviewer: Eu geralmente falo para as pessoas que entrevisto nas etapas de Code Design e Arquitetura que não vamos avaliar a capacidade dela de utilizar a ferramenta ou quão bonito o desenho fica, mas somente o caminho que ela leva até chegar a solução. Esse ponto me parece ir um pouco na contra-mão disso, acho que pode gerar ansiedade.

| ⚠️ | Codificação defensiva razoável. Verifique se há nulos, coleções vazias, etc. Pode omitir se a validade da entrada foi esclarecida com o entrevistador. |
| ❌ | Fique em silêncio enquanto codifica. |
| ❌ | Gaste muito tempo escrevendo comentários. |
| ❌ | Copie e cole o código sem verificar (por exemplo, as variáveis ​​precisam ser renomeadas). |
| ❌ | Interrompa a pessoa entrevistadora quando ela estiver falando. Normalmente, se elas falam, estão tentando dar dicas ou guiar você na direção certa. |

### Após a codificação

| | Coisas |
| --- | --- |
| ✅ | Verifique se há erros em seu código, como se fosse a primeira vez que você via um código escrito por outra pessoa. |

> | ✅ | Crie casos de teste. Experimente casos de teste extremos - conjuntos vazios, conjuntos de item único, números negativos |

Reviewer 3: Pela minha experiência, quase nunca da tempo da pessoa realmente codar os testes. O que rola é mais um papo sobre como, e o que, a pessoa testaria.

Author: Acho que ainda é válido manter esse ponto. Podemos reformular a frase para algo do tipo: Explore casos de teste com o entrevistador. Discorra sobre casos de teste extremos - conjuntos vazios, conjuntos de item único, números negativos

| ✅ | Percorra seu código com esses casos de teste. |
| ✅ | Procure lugares onde você pode refatorar. |

> | ✅ | Caso sinta-se a vontade e confiante, mencione sobre a complexidade de tempo e espaço de seu código. |

Reviewer: ⭐

> | ✅ | Caso sinta-se a vontade e confiante, mencione sobre a complexidade de tempo e espaço de seu código. |

Reviewer 2: ```suggestion
| ✅ | Caso sinta-se à vontade e confiante, mencione sobre a complexidade de tempo e espaço de seu código. |
```

> | ✅ | Explique as compensações e como o código / abordagem pode ser melhorado se tiver mais tempo. |

Reviewer 2: ```suggestion
| ✅ | Explique as análises de trade-off e como o código / abordagem poderia ser melhorado se você tivesse mais tempo. |
```

| ❌ | Anuncie imediatamente que você concluiu a codificação. Faça o anterior primeiro! |

### Finalizando

| | Coisas |
| --- | --- |
| ✅ | Pergunte. Mais importante ainda, faça perguntas boas e envolventes, feitas sob medida para a empresa! |

> | ❌  | Pergunte sobre o seu desempenho na entrevista. Sempre temos mais de uma pessoa entrevistando, ao término da entrevista essas pessoas se reúnem e trocam impressões para gerar um feedback. Esse feedback é consolidado com os das demais etapas e enviado a você, prometemos que fazemos isso rapidamente! |

Reviewer 2: ```suggestion
| ❌  | Pergunte sobre o seu desempenho na entrevista. Você receberá seu feedback compilado posteriormente, após juntarmos as impressões de todas as entrevistas aplicadas. Prometemos que fazemos isso rapidamente! |
```

### Após a entrevista

| | Coisas |
| --- | --- |
| ✅ | Anote as perguntas e respostas da entrevista, pois podem ser úteis para referência futura. |

> | ⚠️ | Envie um e-mail de acompanhamento ao (s) seu (s) entrevistador (es) agradecendo pelo tempo dispensado e pela oportunidade de entrevistá-los. |

Reviewer 2: O final ficou meio esquisito ("entrevistá-los" parece partir do ponto de vista do candidato). Sugiro só remover essa linha, mesmo.
```suggestion
```
